app/web/src/store/realtime/heimdall.ts

```diff
@@ -182,16 +182,17 @@ const updateCache = (
   if (!removed && !data) return;
 
   queryClient.setQueryData(queryKey, (cachedData: { id: string }[]) => {
-    // Don't mutate if we haven't got any data yet
-    // TODO it seems wrong not to add items to an empty list ...
-    if (!cachedData || cachedData.length === 0) {
-      return cachedData;
+    if (!cachedData) {
+      return;
     }
 
     if (removed) {
       // Filter out the item if it is removed
       return cachedData.filter((d) => d?.id !== id);
     } else {
+      if (cachedData.length === 0) {
```

Comment:
You should be able to remove this if statement entirely: if the list is empty, we'll go to the `else` case and it will do the same thing!